### PR TITLE
Create release with optional name and message when generate_release_notes is true

### PR DESCRIPTION
**Pull request by heitorpolidoro**
Name and Message are not required when generate_release_notes it true in the API: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release

Broke if you want to use generate_release_notes without passing name or message to generate automatically

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -1207,10 +1207,10 @@ def create_git_tag_and_release(
         self,
         tag: str,
         tag_message: str,
-        release_name: str,
-        release_message: str,
         object: str,
         type: str,
+        release_name: Opt[str] = NotSet,
```
Comment (EnricoMi) [inline github/Repository.py:1212]: This is a breaking change, but I cannot see how to avoid that...

Comment (EnricoMi) [reply github/Repository.py:1212]: Please adjust the order of the `:param`s in the docstring.

File: github/Repository.py [hunk 2]
```diff
@@ -1264,8 +1264,8 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert is_optional(name, str), name
```
Comment (EnricoMi) [inline github/Repository.py:1267]: Should we consider the value of `generate_release_notes`? As you said, it is optional only if `generate_release_notes` is true.

File: github/Repository.py [hunk 3]
```diff
@@ -1273,8 +1273,8 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert generate_release_notes and is_optional(name, str) or isinstance(name, str), name
+        assert generate_release_notes and is_optional(message, str) or isinstance(message, str), message
         assert isinstance(draft, bool), draft
         assert isinstance(prerelease, bool), prerelease
         assert isinstance(generate_release_notes, bool), generate_release_notes
```
Comment (EnricoMi) [inline github/Repository.py:1280]: I think if we swap the `or` elemens, this reads more clearly:
```suggestion
        assert isinstance(generate_release_notes, bool), generate_release_notes
        assert isinstance(name, str) or generate_release_notes and is_optional(name, str), name
        assert isinstance(message, str) or generate_release_notes and is_optional(message, str), message
        assert isinstance(draft, bool), draft
        assert isinstance(prerelease, bool), prerelease
```
And we need to assert `generate_release_notes` before using it as a `bool`.

File: github/Repository.py (hunk 1 above)
Comment (EnricoMi) [reply github/Repository.py:1221]: I am afraid we have to revert this breaking change. Please replace
```
        release_name: str,
        release_message: str,
```
with
```
        release_name: Opt[str],
        release_message: Opt[str],
```
instead.

Calling code will have to explicitly provide `release_name=None, release_message=None`.

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Happy to approve, please see
- https://github.com/PyGithub/PyGithub/pull/2868/files#r1497080554
- https://github.com/PyGithub/PyGithub/pull/2868/files#r1497083233

**Comment by heitorpolidoro** [comment]
Sorry for the delay in fixing

File: github/Repository.py [hunk 4]
```diff
@@ -1273,23 +1274,27 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert isinstance(generate_release_notes, bool), generate_release_notes
+        assert isinstance(name, str) or generate_release_notes and (name is None or is_optional(name, str)), name
+        assert (
+            isinstance(message, str) or generate_release_notes and (message is None or is_optional(message, str))
+        ), message
```
Comment (EnricoMi) [inline github/Repository.py:1281]: This project doesn't use `None` for unset optional arguments but `NotSet`. Staying with the convention:
```suggestion
        assert isinstance(name, str) or generate_release_notes and is_optional(name, str), name
        assert (
            isinstance(message, str) or generate_release_notes and is_optional(message, str)
        ), message
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Does this work for you?

**Pull request merged**